client/state/atomic/software/reducer.js

```diff
@@ -11,6 +11,14 @@ function software( state = {}, action ) {
 				softwareSet: action.softwareSet,
 				...action?.status,
 			};
+
+		case ATOMIC_SOFTWARE_SET_ERROR:
+			return {
+				...state,
+				siteId: action.siteId,
+				softwareSet: action.softwareSet,
+				error: action.error,
+			};
```

Comment:
This needs to clear out previous status values.

Reply:
Can leave this for now, handling in https://github.com/Automattic/wp-calypso/pull/59394